test-kit/src/test/scala/rating/glicko/GlickoCalculatorWithColorAdvantageTest.scala

```diff
@@ -0,0 +1,308 @@
+package chess.rating.glicko
+
+import chess.{ ByColor, Outcome }
+import munit.ScalaCheckSuite
+
+class GlickoCalculatorWithColorAdvantageTest extends ScalaCheckSuite with chess.MunitExtensions:
+  // Validate results with reference implementations
+  // http://www.glicko.net/glicko/glicko2.pdf
+  val R: Double = 1500d
+  val RD: Double = 350d
+  val V: Double = 0.06d
+
+  val calc = GlickoCalculator(
+    ratingPeriodsPerDay = RatingPeriodsPerDay(0.21436d),
+    colorAdvantage = ColorAdvantage.zero
+  )
+
+  val calcWithAdvantage = GlickoCalculator(
+    ratingPeriodsPerDay = RatingPeriodsPerDay(0.21436d),
+    colorAdvantage = ColorAdvantage.standard
+  )
+
+  val calcWithMoreAdvantage = GlickoCalculator(
+    ratingPeriodsPerDay = RatingPeriodsPerDay(0.21436d),
+    colorAdvantage = ColorAdvantage.crazyhouse
+  )
+
+  def computeGame(players: ByColor[Player], outcome: Outcome) =
+    calc.computeGame(Game(players, outcome), skipDeviationIncrease = true).get.toPair
+
+  def computeGameWithAdvantage(players: ByColor[Player], outcome: Outcome) =
+    calcWithAdvantage.computeGame(Game(players, outcome), skipDeviationIncrease = true).get.toPair
+
+  def computeGameWithMoreAdvantage(players: ByColor[Player], outcome: Outcome) =
+    calcWithMoreAdvantage.computeGame(Game(players, outcome), skipDeviationIncrease = true).get.toPair
+
+  {
+    val players = ByColor.fill:
+      Player(
+        Glicko(rating = R, deviation = RD, volatility = V)
+      )
+    test("default deviation: white wins"):
+      val (w1, b1) = computeGame(players, Outcome.white)
+      val (w2, b2) = computeGameWithAdvantage(players, Outcome.white)
+      val (w3, b3) = computeGameWithMoreAdvantage(players, Outcome.white)
+      assertCloseTo(w1.rating, 1662.21d, 0.005d)
+      assertCloseTo(b1.rating, 1337.79d, 0.005d)
+      // outcome was equally probable for both players
+      assert(w1.deviation == b1.deviation)
+      assertCloseTo(w1.deviation, 290.2305d, 0.00005d)
+      assertCloseTo(b1.deviation, 290.2305d, 0.00005d)
+      // outcome was equally probable for both players
+      assert(w1.volatility == b1.volatility)
+      assertCloseTo(w1.volatility, 0.0599993d, 0.0000001d)
+      assertCloseTo(b1.volatility, 0.0599993d, 0.0000001d)
+      assert(w2.deviation == b2.deviation)
+      assertCloseTo(w2.deviation, 290.2407d, 0.00005d)
+      assertCloseTo(b2.deviation, 290.2407d, 0.00005d)
+      assert(w2.volatility == b2.volatility)
+      assertCloseTo(w2.volatility, 0.0599993d, 0.0000001d)
+      assertCloseTo(b2.volatility, 0.0599993d, 0.0000001d)
+      assert(w3.deviation == b3.deviation)
+      assertCloseTo(w3.deviation, 290.2692d, 0.00005d)
+      assertCloseTo(b3.deviation, 290.2692d, 0.00005d)
+      assert(w3.volatility == b3.volatility)
+      assertCloseTo(w3.volatility, 0.0599992d, 0.0000001d)
+      assertCloseTo(b3.volatility, 0.0599992d, 0.0000001d)
+      assert(w1.rating > w2.rating)
+      assert(b1.rating < b2.rating)
```

Comment:
isn't the most useful `assertCloseTo` about `w2.rating` and `w3.rating`? The whole point is to see how they're different from `w1.rating` due to the advantage.

But I don't see any assertion about the white or black rating of any game with advantage.

Reply:
Ah, that's correct.  I've added tests for these values, so now all values use `assertCloseTo` tests.